File: WoofIOS/SwiftPackages/Presentation/Grid/Sources/PresentationGrid/UIModel/DogCellUIModel.swift
import Models

public struct DogCellUIModel: Hashable, Identifiable {
    public let index: Int
    public let id: Int
    public let name: String
    public let photoUrl: String
    let domain: Dog

    init(index: Int, id: Int, name: String, photoUrl: String, domain: Dog) {
        self.index = index
        self.id = id
        self.name = name
        self.photoUrl = photoUrl
        self.domain = domain
    }
}

File: WoofAndroid/presentation/grid/src/main/kotlin/com/perrystreet/woof/presentation/grid/uimodel/DogCellUIModel.kt
package com.perrystreet.woof.presentation.grid.uimodel

import com.perrystreet.woof.models.dog.Dog

data class DogCellUIModel(
    val index: Int,
    val id: Long,
    val name: String,
    val photoUrl: String,
    internal val domain: Dog,
)
